addons/mail/models/mail_activity.py

```diff
@@ -328,6 +328,8 @@ def create(self, values):
             self.env['bus.bus'].sendone(
                 (self._cr.dbname, 'res.partner', activity.user_id.partner_id.id),
                 {'type': 'activity_updated', 'activity_created': True})
+        if activity.res_model and activity.res_id:
+            self.env[activity.res_model].browse(activity.res_id).set_exception_activity_type()
```

Comment:
1. We could avoid to set_exception_activity_type if activity_type_id.decoration_type not in ['warning', 'danger'] in create.
-> create only, write and unlink need to set_exception_activity_type too if we remove a warning/danger activity type.